src/query/service/tests/it/tests/sessions.rs

```diff
@@ -512,10 +512,8 @@ impl SingletonImpl<Arc<Config>> for TestGlobalServices {
     fn init(&self, value: Arc<Config>) -> Result<()> {
         match std::thread::current().name() {
             None => panic!("thread name is none"),
-            Some(name) => match self.query_config.lock().entry(name.to_string()) {
-                Entry::Vacant(v) => v.insert(value),
-                Entry::Occupied(_v) => panic!("GlobalConfig set twice in test[{:?}]", name),
-            },
+            // allow overwrite config in test.
```

Comment:
Config should not be overwritten, it must be buggy.

Reply:
The unit test `test_abort_on_error` will init multiple times.

Reply:
The unit test should handle the case. This why I comment above we should add a unit test.

Reply:
> The unit test `test_abort_on_error` will init multiple times and will lead to panic.

The test should be refactored.